### PR TITLE
Explicitly build ts docs

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -34,10 +34,13 @@ jobs:
         if: ${{ steps.build-cache.outputs.cache-hit != 'true' }}
         run: yarn install --frozen-lockfile --prefer-offline
 
-      - name: build
+      - name: Build
         if: ${{ steps.build-cache.outputs.cache-hit != 'true' }}
         run: yarn build
 
+      - name: Generate docs
+        run: turbo run docs
```
Comment (TheSonOfThomp) [inline .github/workflows/release.yml:42]: We're caching the `dist` directory (L20), so this file won't get picked up in the cache restore. 
We should add `tsdoc.json` and `stories.js` to the cache path on L20, and L106

**Review by TheSonOfThomp — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (bruugey) [reply .github/workflows/release.yml:42]: I'm confused bc this says rely on the build step

Comment (bruugey) [reply .github/workflows/release.yml:42]: So we add it to the build step

Comment (bruugey) [reply .github/workflows/release.yml:42]: And run it without condition

Comment (bruugey) [reply .github/workflows/release.yml:42]: Should we not check for a build cache in the build step when releasing, if we know we want the build to look different when releasing than when it's just a normal PR

**Review by TheSonOfThomp — APPROVED** [review]

**Pull request merged**